RecoLocalCalo/EcalRecProducers/plugins/alpaka/TimeComputationKernels.h

```diff
@@ -0,0 +1,1169 @@
+              chi2s[ch] = chi2;
+              sum0s[ch] = sum0;
+              sumAAs[ch] = sumAA;
+
+#ifdef DEBUG_TC_NULLHYPOT
+              if (ch == 0) {
+                printf("chi2 = %f sum0 = %d sumAA = %f\n", chi2, static_cast<int>(sum0), sumAA);
+              }
+#endif
+            }
+          }
+        }
+      };
+
+      //
+      // launch ctx parameters are
+      // 45 threads per channel, X channels per block, Y blocks
+      // 45 comes from: 10 samples for i <- 0 to 9 and for j <- i+1 to 9
+      // TODO: it might be much beter to use 32 threads per channel instead of 45
+      // to simplify the synchronization
+      class kernel_time_compute_makeratio {
+        using ScalarType = ::ecal::multifit::SampleVector::Scalar;
+
+      public:
+        template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+        ALPAKA_FN_ACC void operator()(TAcc const& acc,
+                                      EcalDigiDeviceCollection::ConstView digisDevEB,
+                                      EcalDigiDeviceCollection::ConstView digisDevEE,
+                                      ScalarType* const sample_values,
+                                      ScalarType* const sample_value_errors,
+                                      bool* const useless_sample_values,
+                                      char* const pedestal_nums,
+                                      ScalarType* const sumAAsNullHypot,
+                                      ScalarType* const sum0sNullHypot,
+                                      ScalarType* tMaxAlphaBetas,
+                                      ScalarType* tMaxErrorAlphaBetas,
+                                      ScalarType* g_accTimeMax,
+                                      ScalarType* g_accTimeWgt,
+                                      TimeComputationState* g_state,
+                                      EcalMultifitParametersDevice::ConstView paramsDev,
+                                      ConfigurationParameters::type const timeFitLimits_firstEB,
+                                      ConfigurationParameters::type const timeFitLimits_firstEE,
+                                      ConfigurationParameters::type const timeFitLimits_secondEB,
+                                      ConfigurationParameters::type const timeFitLimits_secondEE) const {
+          // constants
+          constexpr uint32_t nchannels_per_block = 10;
+          constexpr auto nthreads_per_channel = nchannels_per_block * (nchannels_per_block - 1) / 2;
+          constexpr auto nsamples = EcalDataFrame::MAXSAMPLES;
+          auto const nchannels = digisDevEB.size() + digisDevEE.size();
+          auto const offsetForInputs = digisDevEB.size();
+          auto const totalElements = nthreads_per_channel * nchannels;
+
+          auto const elemsPerBlock = alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[0u];
+          assert(nthreads_per_channel * nchannels_per_block == elemsPerBlock);
```

Comment:
Eventually the `assert()` needs to be replaced with something else (ideally `ALPAKA_ASSERT_OFFLOAD()`).

Reply:
Added to #43551